Sources/StreamChatUI/CommonViews/AvatarView/ChatPresenceAvatarView/ChatOnlineIndicatorView.swift

```diff
@@ -5,16 +5,22 @@
 import StreamChat
 import UIKit
 
+public protocol MaskProviding {
+    /// Path used to mask space in super view.
+    ///
+    /// No mask is used when nil is returned
+    var maskingPath: CGPath? { get }
+}
+
```

Comment:
I'm not 100% convinced about this approach. We usually use subclassing for cases like this. Would be great to get an opinion from the rest of the team: cc @b-onc @DominikBucher12 @nuno-vieira